controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java

```diff
@@ -133,7 +137,6 @@ public void triggerReadyJobs() {
 
     /** Find the next step to trigger if any, and triggers it */
     public void triggerReadyJobs(LockedApplication application) {
-        if ( ! application.change().isPresent()) return;
```

Comment:
This blocked getting the deployment spec through, so it had to go. It's check again later, however. I was even checked again twice, I believe, before eventual triggering ;)